src/components/Items/ItemEditor.vue

```diff
@@ -0,0 +1,173 @@
+      <!-- question input box -->
+      <input-text
+        :placeholder="'placeholder'"
+        :title="'Question'"
+        v-model:value="questionText"
+        ref="questionText"
+      ></input-text>
+    </div>
+  </div>
+</template>
+
+<script>
+import IconButton from "../UI/Buttons/IconButton.vue";
+import ItemDropDown from "../UI/DropDownMenu/ItemDropDown.vue";
+import InputText from "../UI/Text/InputText.vue";
+
+export default {
+  name: "ItemEditor",
+
+  data() {
+    return {
+      previousItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-left-solid",
+      },
+      nextItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-right-solid",
+      },
+      addItemIconConfig: {
+        enabled: true,
+        iconName: "plus-solid",
+      },
+      deleteItemIconConfig: {
+        enabled: true,
+        iconName: "delete",
+      },
+      selectedItemIndex: 0,
+    };
+  },
+
+  props: {
+    itemList: {
+      default: () => [],
+      type: Array,
+    },
+  },
+
+  components: {
+    ItemDropDown,
+    IconButton,
+    InputText,
+  },
+
+  created() {},
+
+  methods: {
+    capitalizeFirstLetter(str) {
+      return str.charAt(0).toUpperCase() + str.slice(1);
+    },
+    updateSelectedItemIndex(index) {
+      this.selectedItemIndex = index;
+    },
+  },
+
+  computed: {
+    localItemList: {
+      // a local copy of the item list
+      get() {
+        return this.itemList;
+      },
+      set(localItemList) {
+        this.$emit("update:itemList", localItemList);
+      },
+    },
+    itemOptionsList() {
+      // preparing an options list to pass to the dropdown
+      // TODO: ditch the primevue implementation
+      var optionsList = [];
+      this.localItemList.forEach((item, itemIndex) => {
+        var currentItem = {};
+        currentItem["index"] = itemIndex;
+        var itemType = this.capitalizeFirstLetter(item.type);
+        currentItem["label"] = `${itemType} ${itemIndex + 1}`;
+        optionsList.push(currentItem);
+      });
+
+      return optionsList;
+    },
+    isLastItem() {
+      // is the current selected item the last item in the list?
+      return this.selectedItemIndex == this.localItemList.length - 1;
+    },
+    isFirstItem() {
+      // is the current selected item the first item in the list?
+      return this.selectedItemIndex == 0;
+    },
+    questionText: {
+      get() {
+        // extract question text from item
+        return this.localItemList[this.selectedItemIndex].details.text;
+      },
+      set(value) {
+        // set the updated question text back into the item
+        this.localItemList[this.selectedItemIndex].details.text = value;
+      },
+    },
+  },
+
+  emits: ["update:itemList"],
+};
+</script>
+
+<style scoped>
+.nav-bar {
+  height: 10%;
+}
+.main-container {
+  background: #f4eae1;
+}
+.item-editor-box {
+  background: #fff6ef;
+}
+.editor-title {
+  color: #bcafa5;
+}
```

Comment:
would be better to have these as custom classes in tailwind?

Reply:
That is the plan, keeping it here and will move to custom classes when we'll be finishing up the css + styling